### ISSUE TITLE
[Bug]: dbengine data retention not working as expected

### ISSUE THREAD

**Issue by mpohle-transporeon** [netdata/netdata#13557]
### Bug description

Hi everbody,
since one of the latest updates of netdata old data is not deleted as configured.

We configured the following settings in the netdata config:

```
[global]
bind socket to IP = *
run as user = netdata
dbengine multihost disk space = 100

[health]
enabled = no

[db]
mode = dbengine
dbengine multihost disk space MB = 100

[ml]
enabled = no
```

On disk with this configuration there are almost 2GB used.
```
$ du -sh /var/cache/netdata/*
1.9G    /var/cache/netdata/dbengine
922M    /var/cache/netdata/netdata-meta.db
32K     /var/cache/netdata/netdata-meta.db-shm
6.4M    /var/cache/netdata/netdata-meta.db-wal
```

### Expected behavior

dbengine should not use more than 100MB.

### Steps to reproduce

Will happen without doing anything after some days/weeks.

### Installation method

native binary packages (.deb/.rpm)

### System info

```shell
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 10 (buster)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="10"
/etc/os-release:VERSION="10 (buster)"
/etc/os-release:VERSION_CODENAME=buster
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Version: netdata v1.35.1
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         NO
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
```

**Comment by MrZammler** [comment]
Hi!

Can you please send the output of `http://localhost:19999/netdata.conf` to `manolis@netdata.cloud` ? Also if possible the `error.log` ?

There is a new version (1.36.1) which you can check, there where some changes to the config entries during that time, not sure though if it's related.

Thanks!

**Comment by MrZammler** [comment]
I can't see something really wrong in the logs (thanks for sending!), but I will continue to look. It seems to be deleting journal and datafiles as it creates new ones. I will try to replicate with 1.35.1

It appears that at some point you deleted the contents of the `dbengine` directory and restarted? Is it possible to update to 1.36.1 and continue to monitor it?

**Comment by MrZammler** [comment]
I don't think 1.36.1 should have any fix in this regard...

Can you please check from `http://localhost:19999/api/v1/charts` the values of `charts_count` and `dimensions_count` ? Also, your error.log does not indicate as such, but just to make sure, is this agent configured as a parent, i.e. other agents streaming to it?

Thanks!

**Comment by mpohle-transporeon** [comment]
"charts_count": 562,
"dimensions_count": 2955,

No the agent is standalone and not configured to send data to somewhere else.

**Comment by MrZammler** [comment]
Just an update on this, from the following chart:

![image](https://user-images.githubusercontent.com/1905463/188084061-fe42b09e-6755-495f-b487-24ad8803a750.png)

It seems used_by_collectors is constantly growing, which is used when testing dbengine quota.

It could be a runaway collector, or something in the system that is constantly creating new dimensions (without actually collecting them though).

**Comment by Dim-P** [comment]
@mpohle-transporeon Any news on this?

**Comment by MrZammler** [comment]
Hi @mpohle-transporeon 

We're closing this for now, please let us know if this continues to be an issue for you or it was resolved.

Thanks!

**Comment by elisdg** [comment]
Hi @MrZammler,
sorry I didn't follow up on this. The collegue moved and I didn't have the time yet to follow up, but as this is still bugging, even with the current version 1.37.1 I would be thankful, if we could look into this again.

I tryed to set the config but it didn't seem to help:
```
[plugin]
cgroups=no
```

I don't know if relevant, but observations on the affected VMs are, that may be different to the other machines where problems aren't visible:
- systemctl status netdata shows: memory  > 500M
- there are docker containers running? partly multiple. or multiple images? It "feels like" the more containers are (not necessarily running) there, the faster the cache directory gets filled?!

**Comment by vboerner-tpgroup** [comment]
Hi @MrZammler,
sorry I didn't follow up on this. The collegue moved and I didn't have the time yet to follow up, but as this is still bugging, even with the current version 1.37.1 I would be thankful, if we could look into this again.

I tryed to set the config but it didn't seem to help:
```
[plugin]
cgroups=no
```

I don't know if relevant, but observations on the affected VMs are, that may be different to the other machines where problems aren't visible:
- systemctl status netdata shows: memory  > 500M
- there are docker containers running? partly multiple. or multiple images? It "feels like" the more containers are (not necessarily running) there, the faster the cache directory gets filled?!
- after (re)starting the service there are messages like:
`Does not have a configuration file inside /etc/netdata/ebpf.d.conf. It will try to load stock file.` and `Your environment does not have BTF file /sys/kernel/btf//vmlinux. The plugin will work with 'legacy' code` and `Cannot read process groups configuration file '/etc/netdata/apps_groups.conf'. Will try '/usr/lib/netdata/conf.d/apps_groups.conf'` and `Cannot open file '/proc/7264/status'` or things alike

**Comment by vboerner-transporeon** [comment]
UPDATE:
I get the impression, that it has something to do with **restarting** containers
There was a machine (debian10) in the morning that reported low memory. When I looked at it, netdata (v1.37.1) process consumed 50% memory, after restarting the service the process was at 10% (but even that seems a little high)
There are running 3 containers and one is constantly restarting (unfortunately I don't know how long this is the case)

I'm trying to reproduce this on other machines where an nginx container is running.
Executing `watch -n20 'docker restart nginx'` I can observer that the **/var/cache/netdata** directory is constantly growing. (memory consumpting of netdata process on those machines is currently at arount 21%)

**Machine1**: /var/cache/netdata after restart the service at ~10:05AM: 474008K ---> now at ~11:55AM: 505588K
sidenote: I didn't clean the directory after restarting the service. The size didn't drop to below 100K like in the config:
```
[global]                               
bind socket to IP = *                  
run as user = netdata                  
dbengine multihost disk space = 100    
                                       
[health]                               
enabled = no                           
                                       
[db]                                   
mode = dbengine                        
dbengine multihost disk space MB = 100 
```
Machine1 also ran a job in the morning where hundreds of containers are started, executing a chef:inspec job, and then removed.
**Machine 2:**
/var/cache/netdata after running the service since 2022-12-28:  1085784K ---> now at ~11:55AM: 1096384K
(this machine only runs the `watch -n 30 'docker restart nginx'` command)

**Comment by vboerner-transporeon** [comment]
I was just observing, that we have a k8s machine (with containerd) running AWX (so, lots of containers...) thats presenting similar issue. (this special machine runs Ubuntu 20.04 and was set into action in the last month or weeks, so no long-time experience there)

**Comment by ilyam8** [comment]
Indeed, there was an issue with dbengine disk space usage. The issue is fixed in the master branch. @vboerner-transporeon consider updating your Netdata's.

**Comment by vboerner-transporeon** [comment]
> Indeed, there was an issue with dbengine disk space usage. The issue is fixed in the master branch. @vboerner-transporeon consider updating your Netdata's.

sorry for coming back just yet. The issue seems to be solved. Thank you for your work @ilyam8 , very much appreciated!
